### PR TITLE
Fulfill logistics requests for Buffer and Requester Logistic Chests

### PR DESCRIPTION
**Pull request by bengardner**
This tracks logistic-chest-requester and logistic-chest-requester items and fulfills requests from the item network.

### CODE REVIEW

File: src/GlobalState.lua [hunk 1]
```diff
@@ -90,6 +95,36 @@ function M.shuffle(list)
   end
 end
 
+function M.logistic_scan()
+  local entities = game.surfaces[1].find_entities_filtered{name = {"logistic-chest-requester", "logistic-chest-buffer"}}
```
Comment (year6b7a) [inline src/GlobalState.lua]: Should this function loop through every surface?

File: src/GlobalState.lua [hunk 2]
```diff
@@ -90,6 +95,36 @@ function M.shuffle(list)
...
+end
+
+function M.logistic_add_entity(entity)
+  if global.mod.logistic[entity.unit_number] == nil then
+    global.mod.logistic[entity.unit_number] = entity
+    Queue.push(global.mod.scan_queue, entity.unit_number)
+  end
+end
+
+function M.logistic_del_entity(entity)
```
Comment (year6b7a) [inline src/GlobalState.lua]: Is there any difference between this an `M.logicstic_del` besides using entity vs unit_number? Can this be changed to only use `M.logistic_del`?

File: src/GlobalState.lua [hunk 3]
```diff
@@ -90,6 +95,36 @@ function M.shuffle(list)
...
+  if global.mod.logistic[entity.unit_number] ~= nil then
+    global.mod.logistic[entity.unit_number] = nil
+  end
+end
+
+function M.logistic_del(unit_number)
+  if global.mod.logistic[unit_number] == nil then
```
Comment (year6b7a) [inline src/GlobalState.lua]: Should this be `~=`?

File: src/GlobalState.lua (hunk 2 above)
Comment (bengardner) [reply src/GlobalState.lua]: Sure, I assume you'd be OK doing that? Easy enough to change the caller to use `entity.unit_number`

File: src/GlobalState.lua (hunk 3 above)
Comment (bengardner) [reply src/GlobalState.lua]: Yes. Good catch.

File: src/GlobalState.lua (hunk 1 above)
Comment (bengardner) [reply src/GlobalState.lua]: Probably.  I'm new, so I don't really understand multiple surfaces.

**Comment by bengardner** [comment]
I reworked the code to support any container/chest that implements the logistic interface.

**Pull request merged**